src/core/CoolTime.cpp

```diff
@@ -29,231 +29,63 @@
-        break;
-      }
-    }
-    char monthAbbr[4] = {__TIMESTAMP__[4], __TIMESTAMP__[5], __TIMESTAMP__[6],
-                         0};
-    char tempDay[3] = {__TIMESTAMP__[8], __TIMESTAMP__[9], 0};
-    int Day = atoi(&tempDay[0]);
-    char tempHour[3] = {__TIMESTAMP__[posMarker - 2],
-                        __TIMESTAMP__[posMarker - 1], 0};
-    int Hour = atoi(&tempHour[0]);
-    char tempMinute[3] = {__TIMESTAMP__[posMarker + 1],
-                          __TIMESTAMP__[posMarker + 2], 0};
-    int Minute = atoi(&tempMinute[0]);
-    char tempSecond[3] = {__TIMESTAMP__[posMarker + 4],
-                          __TIMESTAMP__[posMarker + 5], 0};
-    int Second = atoi(&tempSecond[0]);
-    char tempYear[3] = {__TIMESTAMP__[posMarker + 9],
-                        __TIMESTAMP__[posMarker + 10], 0};
-    int Year = atoi(&tempYear[0]);
-    int Month;
-
-    if (strstr(monthAbbr, "Jan")) {
-      Month = 1;
-    }
-    if (strstr(monthAbbr, "Feb")) {
-      Month = 2;
-    }
-    if (strstr(monthAbbr, "Mar")) {
-      Month = 3;
-    }
-    if (strstr(monthAbbr, "Apr")) {
-      Month = 4;
-    }
-    if (strstr(monthAbbr, "May")) {
-      Month = 5;
-    }
-    if (strstr(monthAbbr, "Jun")) {
-      Month = 6;
-    }
-    if (strstr(monthAbbr, "Jul")) {
-      Month = 7;
-    }
-    if (strstr(monthAbbr, "Aug")) {
-      Month = 8;
-    }
-    if (strstr(monthAbbr, "Sep")) {
-      Month = 9;
-    }
-    if (strstr(monthAbbr, "Oct")) {
-      Month = 10;
-    }
-    if (strstr(monthAbbr, "Nov")) {
-      Month = 11;
-    }
-    if (strstr(monthAbbr, "Dec")) {
-      Month = 12;
-    }
-    setDateTime(y2kYearToTm(Year), Month, Day, Hour, Minute, Second);
-    unsigned long instantTime = RTC.get(CLOCK_ADDRESS);
-    this->timeSync = instantTime;
-    this->compileTime = false;
-    this->config(false);
-    DEBUG_VAR("RTC set from:", __TIMESTAMP__);
-    DEBUG_VAR("Seconds since UNIX Epoch:", instantTime);
-  }
+extern "C" int clock_gettime(clockid_t unused, struct timespec *tp);
+timeval cbtime;
+timeval tv;
+timespec tp;
+time_t now;
+
+void CoolTime::begin() {
+  configTime(TZ_SEC, DST_SEC, "pool.ntp.org");
 }
 
-void CoolTime::update() {
-  if (this->NTP && WiFi.status() == WL_CONNECTED) {
-    if (!this->isServerSelected()) {
-      this->selectTimeServer();
-      this->printConf();
-    }
-    if (!(this->isTimeSync())) {
-      int repeats = 0;
-
-      DEBUG_LOG("Waiting for sync");
-      this->timeSync = this->getNtpTime();
-      while (!this->timeSync) {
-        delay(1000);
-        this->timeSync = this->getNtpTime();
-
-        if (repeats >= 4) {
-          selectTimeServer();
-          delay(500);
-          this->timeSync = this->getNtpTime();
-          break;
-        }
-        repeats++;
-      }
-      breakTime(this->timeSync, this->tmSet);
-      this->rtc.set(makeTime(this->tmSet), CLOCK_ADDRESS);
-    }
-    this->config(true);
+bool CoolTime::update() {
+  if (WiFi.status() == WL_CONNECTED) {
```

Comment:
But... You've mixed NTP and RTC github issues in the same pull request :cold_sweat: ???

What is the reason behind this ? Are things not complicated enough already ???